addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/handler/SenseBoxHandler.java

```diff
@@ -0,0 +1,374 @@
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_TEMPERATURE_LR));
+                    }
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshJob = scheduler.scheduleAtFixedRate(runnable, 0, thingConfiguration.getRefreshInterval(),
+                TimeUnit.SECONDS);
+    }
+
+    private boolean fetchData() {
+        try {
+            data = CACHE.get(thingConfiguration.getSenseBoxId());
+        } catch (IOException e) {
+            logger.debug("Exception occurred during fetching data: {}", e.getMessage(), e);
+            return false;
+        }
+
+        return true;
+    }
+
+    private final ExpiringCache<String, SenseBoxData> CACHE = new ExpiringCache<>(CACHE_EXPIRY,
+            new ExpiringCache.LoadAction<String, SenseBoxData>() {
+                @Override
+                public SenseBoxData load(String senseBoxId) throws IOException {
+                    SenseBoxData result = reallyFetchDataFromServer(senseBoxId);
+                    return result;
+                }
+            });
+
+    private SenseBoxData reallyFetchDataFromServer(String senseBoxId) {
+        String query = SENSEMAP_API_URL_BASE + "/boxes/" + senseBoxId;
+
+        // the caching layer does not like null values
+        SenseBoxData result = new SenseBoxData();
+
+        try {
+            URL url = new URL(query);
+            URLConnection connection = url.openConnection();
+            String body = IOUtils.toString(connection.getInputStream());
+
+            logger.debug("Fetched Data");
+            logger.debug(body);
+            SenseBoxData parsedData = gson.fromJson(body, SenseBoxData.class);
+
+            // Could perhaps be simplified via triply-nested arrays
+            // http://stackoverflow.com/questions/36946875/how-can-i-parse-geojson-with-gson
+            for (SenseBoxLoc loc : parsedData.getLocs()) {
+                if (loc.getGeometry() != null) {
+                    List<Double> locationData = loc.getGeometry().getData();
+                    if (locationData != null) {
+                        SenseBoxLocation location = new SenseBoxLocation();
+
+                        if (locationData.size() > 0) {
+                            location.setLongitude(locationData.get(0));
+                        }
+
+                        if (locationData.size() > 1) {
+                            location.setLatitude(locationData.get(1));
+                        }
+
+                        if (locationData.size() > 2) {
+                            location.setHeight(locationData.get(2));
+                        }
+
+                        parsedData.setLocation(location);
+                    }
+                }
+            }
+
+            for (SenseBoxSensor sensor : parsedData.getSensors()) {
```

Comment:
Actually, I did. But sometimes, even eclipse throws Exceptions while formatting, so this file might have slipped through the cracks.